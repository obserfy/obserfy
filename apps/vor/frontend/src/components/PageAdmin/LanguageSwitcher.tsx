import { Trans } from "@lingui/macro"
import { useLocalization } from "gatsby-theme-i18n"

import { Box, Flex } from "theme-ui"
import { ReactComponent as GlobeIcon } from "../../icons/globe.svg"
import { ReactComponent as NextIcon } from "../../icons/next-arrow.svg"
import Icon from "../Icon/Icon"
import Typography from "../Typography/Typography"

const LanguageSwitcher = () => {
  const { locale } = useLocalization()

  const switchLanguage = () => {
    window.localStorage.setItem("preferred-lang", locale === "id" ? "en" : "id")
    window.location.replace(`${locale === "id" ? "" : "/id"}/dashboard/admin`)
  }

  return (
    <Flex
      sx={{
        alignItems: "center",
        cursor: "pointer",
        "&:hover": { backgroundColor: "primaryLightest" },
      }}
      p={3}
      onClick={switchLanguage}
      data-cy={locale === "en" ? "switch-indonesian" : "switch-english"}
    >
      <Icon as={GlobeIcon} fill="background" />
      <Box ml={3}>
        <Typography.Body color="textMediumEmphasis" sx={{ fontSize: 0 }}>
          <Trans>Language</Trans>
        </Typography.Body>
        <Typography.Body>
          {locale === "en" ? "English" : "Indonesian"}
        </Typography.Body>
      </Box>
      <Icon as={NextIcon} ml="auto" />
    </Flex>
  )
}

export default LanguageSwitcher
