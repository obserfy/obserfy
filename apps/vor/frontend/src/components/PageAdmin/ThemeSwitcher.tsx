import { t, Trans } from "@lingui/macro"
import { FC } from "react"
import { Box, Flex, useColorMode } from "theme-ui"
import { ReactComponent as DarkModeIcon } from "../../icons/dark-mode.svg"
import { ReactComponent as LightModeIcon } from "../../icons/light-mode.svg"
import { ReactComponent as NextIcon } from "../../icons/next-arrow.svg"
import Icon from "../Icon/Icon"
import Typography from "../Typography/Typography"

const ThemeSwitcher: FC = () => {
  const [colorMode, setColorMode] = useColorMode()
  const isDark = colorMode === "dark"

  const handleThemeChange = () => setColorMode(isDark ? "default" : "dark")

  return (
    <Flex
      p={3}
      onClick={handleThemeChange}
      data-cy={isDark ? "light-switch" : "dark-switch"}
      sx={{
        alignItems: "center",
        cursor: "pointer",
        "&:hover": { backgroundColor: "primaryLightest" },
      }}
    >
      <Icon as={isDark ? DarkModeIcon : LightModeIcon} />

      <Box ml={3}>
        <Typography.Body color="textMediumEmphasis" sx={{ fontSize: 0 }}>
          <Trans>Theme</Trans>
        </Typography.Body>
        <Typography.Body>
          <Trans id={isDark ? t`Dark` : t`Light`} />
        </Typography.Body>
      </Box>

      <Icon as={NextIcon} ml="auto" />
    </Flex>
  )
}

export default ThemeSwitcher
