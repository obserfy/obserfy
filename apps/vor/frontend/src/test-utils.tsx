/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
