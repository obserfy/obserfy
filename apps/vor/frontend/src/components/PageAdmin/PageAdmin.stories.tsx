import { FC } from "react"
import PageAdmin from "./PageAdmin"

export default {
  title: "Core/PageAdmin",
  component: PageAdmin,
  parameters: {
    componentSubtitle: "Just a simple PageAdmin",
  },
}

export const Basic: FC = () => <PageAdmin />
