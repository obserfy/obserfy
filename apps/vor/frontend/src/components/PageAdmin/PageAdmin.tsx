import { t, Trans } from "@lingui/macro"
import { FC } from "react"
import { Box, Button, Card, Flex } from "theme-ui"
import { useGetSchool } from "../../hooks/api/schools/useGetSchool"
import useLogout from "../../hooks/api/useLogout"
import { ReactComponent as BookIcon } from "../../icons/book-open.svg"
import { ReactComponent as CreditCardIcon } from "../../icons/credit-card.svg"
import { ReactComponent as NextIcon } from "../../icons/next-arrow.svg"
import { ReactComponent as PersonIcon } from "../../icons/person.svg"
import { ReactComponent as SendIcon } from "../../icons/send.svg"
import {
  ADMIN_CURRICULUM_URL,
  ADMIN_GUARDIAN_URL,
  ADMIN_INVITE_USER_URL,
  ADMIN_STUDENTS_URL,
  ADMIN_SUBSCRIPTION_URL,
  ADMIN_USERS_URL,
  CLASS_SETTINGS_URL,
  SCHOOL_PROFILE_URL,
} from "../../routes"
import Icon from "../Icon/Icon"
import { Link } from "../Link/Link"
import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import Typography from "../Typography/Typography"
import LanguageSwitcher from "./LanguageSwitcher"
import ThemeSwitcher from "./ThemeSwitcher"

export const PageAdmin: FC = () => (
  <Box sx={{ maxWidth: "maxWidth.sm" }} m="auto" pb={5}>
    <SchoolName />

    <Card variant="responsive" mb={3}>
      <AdminLink to={ADMIN_CURRICULUM_URL} text={t`Curriculum`} />
      <AdminLink to={CLASS_SETTINGS_URL} text={t`Class`} />
      <AdminLink to={ADMIN_STUDENTS_URL} text={t`All Students`} />
      <AdminLink to={ADMIN_GUARDIAN_URL} text={t`All Guardians`} />
    </Card>

    <Card variant="responsive" mb={3}>
      <AdminLink
        icon={BookIcon}
        to={SCHOOL_PROFILE_URL}
        text={t`School Profile`}
      />
      {/* <AdminLink */}
      {/*  icon={FaceIcon} */}
      {/*  to={USER_PROFILE_URL} */}
      {/*  text={t`User Profile`} */}
      {/*  iconFill="text" */}
      {/* /> */}
    </Card>

    <Card variant="responsive" mb={3}>
      <AdminLink
        to={ADMIN_USERS_URL}
        icon={PersonIcon}
        text={t`Users`}
        iconFill="text"
      />
      <AdminLink
        to={ADMIN_INVITE_USER_URL}
        icon={SendIcon}
        text={t`Invite Your Team`}
      />
    </Card>

    <Card variant="responsive" mb={3}>
      <AdminLink
        to={ADMIN_SUBSCRIPTION_URL}
        icon={CreditCardIcon}
        text={t`Plans & Billing`}
      />
    </Card>

    <Card variant="responsive" mb={3}>
      <LanguageSwitcher />
      <ThemeSwitcher />
    </Card>

    <Flex mx={3} sx={{ flexDirection: ["column", "row"] }}>
      <ChangeSchoolButton />
      <LogoutButton />
    </Flex>
  </Box>
)

const ChangeSchoolButton = () => (
  <Link
    to="/choose-school"
    sx={{
      ml: "auto",
      display: "block",
      width: ["100%", "auto"],
      flexShrink: 0,
    }}
  >
    <Button
      variant="outline"
      color="warning"
      mb={3}
      sx={{ width: ["100%", "auto"] }}
    >
      <Trans>Switch school</Trans>
    </Button>
  </Link>
)

const LogoutButton = () => {
  const logout = useLogout()

  return (
    <Button
      variant="outline"
      ml={[0, 2]}
      color="danger"
      onClick={() => logout.mutate()}
      mb={3}
    >
      <Trans>Log Out</Trans>
    </Button>
  )
}

const SchoolName = () => {
  const schoolDetail = useGetSchool()

  if (schoolDetail.isLoading) {
    return (
      <LoadingPlaceholder
        sx={{ width: "10rem", height: 27 }}
        mx={3}
        mb={3}
        mt={4}
      />
    )
  }

  return (
    <Typography.H6 mx={3} mb={3} mt={4} sx={{ height: 27 }}>
      {schoolDetail.data?.name}
    </Typography.H6>
  )
}

const AdminLink: FC<{
  to: string
  icon?: FC
  text: string
  iconFill?: string
}> = ({ icon, text, to, iconFill = "transparent" }) => (
  <Link
    to={to}
    sx={{ display: "block", "&:hover": { backgroundColor: "primaryLightest" } }}
  >
    <Flex p={3} sx={{ alignItems: "center" }}>
      {icon && <Icon as={icon} mr={3} fill={iconFill} color="text" size={20} />}
      <Typography.Body>
        <Trans id={text} />
      </Typography.Body>
      <Icon as={NextIcon} ml="auto" />
    </Flex>
  </Link>
)

export default PageAdmin
