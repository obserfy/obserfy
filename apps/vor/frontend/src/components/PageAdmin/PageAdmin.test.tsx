import { render } from "../../test-utils"
import { Basic } from "./PageAdmin.stories"

describe("PageAdmin", () => {
  it("should render correctly", () => {
    // TODO: This is a temporary fix, make sure to test mock response correctly.
    const response = {
      name: "Loco",
    }
    fetchMock.mockResponse(JSON.stringify(response))
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
