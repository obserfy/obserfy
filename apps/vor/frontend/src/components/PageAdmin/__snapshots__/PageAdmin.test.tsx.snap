// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAdmin should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: auto;
  padding-bottom: 64px;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  margin-right: 16px;
  margin-bottom: 16px;
  margin-top: 32px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 10rem;
  height: 27px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 0;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  margin-left: 0;
  margin-right: 0;
  margin-bottom: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-3 {
    border-radius: 6px;
    margin-left: 16px;
    margin-right: 16px;
  }
}

.emotion-4 {
  display: block;
}

.emotion-4:hover {
  background-color: var(--theme-ui-colors-primaryLightest);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-left: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: 16px;
  color: var(--theme-ui-colors-text);
  fill: transparent;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-29 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-right: 16px;
  color: var(--theme-ui-colors-text);
  fill: var(--theme-ui-colors-text);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-44 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-44:hover {
  background-color: var(--theme-ui-colors-primaryLightest);
}

.emotion-45 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-background);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-46 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
}

.emotion-47 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-size: 13px;
}

.emotion-50 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-50:hover {
  background-color: var(--theme-ui-colors-primaryLightest);
}

.emotion-51 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-53 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-textMediumEmphasis);
  font-size: 13px;
}

.emotion-56 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: 16px;
  margin-right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media screen and (min-width: 40em) {
  .emotion-56 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

.emotion-57 {
  margin-left: auto;
  display: block;
  width: 100%;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

@media screen and (min-width: 40em) {
  .emotion-57 {
    width: auto;
  }
}

.emotion-58 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-bottom: 16px;
  color: var(--theme-ui-colors-warning);
  width: 100%;
}

.emotion-58:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-58:hover,
.emotion-58:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-58:hover:disabled,
.emotion-58:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

@media screen and (min-width: 40em) {
  .emotion-58 {
    width: auto;
  }
}

.emotion-59 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  margin-left: 0;
  margin-bottom: 16px;
  color: var(--theme-ui-colors-danger);
}

.emotion-59:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-59:hover,
.emotion-59:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-59:hover:disabled,
.emotion-59:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

@media screen and (min-width: 40em) {
  .emotion-59 {
    margin-left: 8px;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
      <div
        class="emotion-3"
      >
        <a
          class="emotion-4"
          href="/dashboard/admin/curriculum"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Curriculum
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
        <a
          class="emotion-4"
          href="/dashboard/admin/class"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Class
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
        <a
          class="emotion-4"
          href="/dashboard/admin/students"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              All Students
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
        <a
          class="emotion-4"
          href="/dashboard/admin/guardians"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              All Guardians
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
      </div>
      <div
        class="emotion-3"
      >
        <a
          class="emotion-4"
          href="/dashboard/admin/school-profile"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-23"
            />
            <p
              class="emotion-6"
            >
              School Profile
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
      </div>
      <div
        class="emotion-3"
      >
        <a
          class="emotion-4"
          href="/dashboard/admin/users"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-29"
            />
            <p
              class="emotion-6"
            >
              Users
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
        <a
          class="emotion-4"
          href="/dashboard/admin/inviteUser"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-23"
            />
            <p
              class="emotion-6"
            >
              Invite Your Team
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
      </div>
      <div
        class="emotion-3"
      >
        <a
          class="emotion-4"
          href="/dashboard/admin/subscription"
        >
          <div
            class="emotion-5"
          >
            <div
              class="emotion-23"
            />
            <p
              class="emotion-6"
            >
              Plans & Billing
            </p>
            <div
              class="emotion-7"
            />
          </div>
        </a>
      </div>
      <div
        class="emotion-3"
      >
        <div
          class="emotion-44"
          data-cy="switch-indonesian"
        >
          <div
            class="emotion-45"
          />
          <div
            class="emotion-46"
          >
            <p
              class="emotion-47"
            >
              Language
            </p>
            <p
              class="emotion-6"
            >
              English
            </p>
          </div>
          <div
            class="emotion-7"
          />
        </div>
        <div
          class="emotion-50"
          data-cy="dark-switch"
        >
          <div
            class="emotion-51"
          />
          <div
            class="emotion-46"
          >
            <p
              class="emotion-53"
            >
              Theme
            </p>
            <p
              class="emotion-6"
            >
              Light
            </p>
          </div>
          <div
            class="emotion-7"
          />
        </div>
      </div>
      <div
        class="emotion-56"
      >
        <a
          class="emotion-57"
          href="/choose-school"
        >
          <button
            class="emotion-58"
          >
            Switch school
          </button>
        </a>
        <button
          class="emotion-59"
        >
          Log Out
        </button>
      </div>
    </div>
  </div>
</div>
`;
